src/searchForm/SearchForm.tsx

```diff
@@ -54,18 +49,18 @@ enum Elements {
 }
 
 type FormValues = {
-  [key in keyof typeof Elements]?: string | number | boolean
+  [key in keyof typeof Elements]?: string | number | boolean | AutocompleteOnChange
 }
 
 const SearchForm = ({
   className,
   onSubmit,
-  autocompleteFromProps,
-  autocompleteToProps,
+  autocompleteFrom,
```

Comment:
it could be nice to add some assertions to make sure this autocomplete are really autocompletes and not some ReactElements.

Reply:
There is a way:

```tsx
// Import the component (the function)
import AutoComplete from 'autoComplete'

// React elements have a `type` attribute referencing either the native HTML tag (`'span'`, `'div'`)
// or the component (`function() { /* ... */ }`)
const isAutoComplete = autocompleteFrom.type === AutoComplete
```

The issue is that you can't compose the component any more:

```tsx
<SearchForm
  autocompleteFrom={(
    <span>
      <AutoComplete />
    <span>
  )}
/>
```

Nor extend it:


```tsx
<SearchForm autocompleteFrom={<AutoCompleteVariant />} />
```

Reply:
Thanks, interesting. It won't work for us then because we compose the AutoComplete in the SPA with another component